### ISSUE TITLE
Can't install: Incorrect arguments passed to the C compiler

### ISSUE THREAD

**Issue by adament** [ueno/ruby-gpgme#67]
I tried installing gpgme 2.0.10 on my Arch Linux machine and got the error:

```
configure: error: in `/tmp/test/gem/gems/gpgme-2.0.10/ext/gpgme/tmp/x86_64-unknown-linux-gnu/ports/libgpg-error/1.20/libgpg-error-1.20':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

Looking in config.log I get:

```
configure:3699: $? = 0
configure:3688: gcc -v >&5
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-unknown-linux-gnu/5.2.0/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: /build/gcc-multilib/src/gcc-5.2.0/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --enable-libmpx --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --with-default-libstdcxx-abi=gcc4-compatible
Thread model: posix
gcc version 5.2.0 (GCC) 
configure:3699: $? = 0
configure:3688: gcc -V >&5
gcc: error: unrecognized command line option '-V'
gcc: fatal error: no input files
compilation terminated.
configure:3699: $? = 1
configure:3688: gcc -qversion >&5
gcc: error: unrecognized command line option '-qversion'
gcc: fatal error: no input files
compilation terminated.
configure:3699: $? = 1
configure:3719: checking whether the C compiler works
configure:3741: gcc '-fPIC '   conftest.c  >&5
gcc: error: '-fPIC: No such file or directory
gcc: error: ': No such file or directory
configure:3745: $? = 1
configure:3783: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "libgpg-error"
| #define PACKAGE_TARNAME "libgpg-error"
| #define PACKAGE_VERSION "1.20"
| #define PACKAGE_STRING "libgpg-error 1.20"
| #define PACKAGE_BUGREPORT "http://bugs.gnupg.org"
| #define PACKAGE_URL ""
| #define PACKAGE "libgpg-error"
| #define VERSION "1.20"
| /* end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:3788: error: in `/tmp/test/gem/gems/gpgme-2.0.10/ext/gpgme/tmp/x86_64-unknown-linux-gnu/ports/libgpg-error/1.20/libgpg-error-1.20':
configure:3790: error: C compiler cannot create executables
See `config.log' for more details
```

Running gcc --version I get

```
gcc (GCC) 5.2.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

I think the error is that the single quote ' is prepended to -fPIC in ext/gpgme/extconf.rb. It works when compiling with --use-system-libraries.

The full command that I ran was:

```
> gem install --ignore-dependencies --no-user-install -i "test/gem" -n "test/usr/bin" gpgme-2.0.10.gem
```
and I have attached the output and the config.log here: https://gist.github.com/adament/68897817542127fc3c93

**Comment by ueno** [comment]
thanks for reporting that. I guess the single quote handling in mini_portile has changed, as I can't reproduce it with 0.6.2, but with 0.7.0.rc4. I will add a work around and update the gem soon.

**Comment by amosshapira** [comment]
In case this helps, I just hit exactly the same problem on OS X 10.10.5 + xcode 7.0.1 + ruby 2.13p242 (installed with rvm) and ruby gpgme 2.0.9 or 2.0.10 (current latest).

Thanks.

**Comment by Hufschmidt** [comment]
The same happens quite reproduceably on a (clean) Debian 8.2 Jessie. Got it runnng using amosshapira 's workaround, thanks.

**Comment by ueno** [comment]
should be fixed in 2.0.11. sorry for the delay.
